tests/compile-fail/fn_ptr_comparison.rs

```diff
@@ -0,0 +1,11 @@
+fn f() -> i32 {
+    42
+}
+
+fn return_fn_ptr() -> fn() -> i32 {
+    f
+}
+
+fn main() {
+    assert!(return_fn_ptr() == f); //~ ERROR assertion failed
```

Comment:
What is the purpose of this test? We already test elsewhere that `assert!(false)` fails, don't we?

Reply:
I just wanted to make sure we're getting the right error. But I guess the run-pass test kinda guarantees that